src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -79,6 +79,10 @@ public Task<VisualStudioProject> CreateAndAddToWorkspaceAsync(string projectSyst
             // From this point on, we start mutating the solution.  So make us non cancellable.
             cancellationToken = CancellationToken.None;
 
+            // Move the rest of this work off from the UI thread; although it should be fast, sometimes an accidental MEF part creation
```

Comment:
📝 You should change `EnsureDocumentOptionProvidersInitializedAsync` above to use `ConfigureAwait(false)`, or `ConfigureAwaitRunInline()`. Otherwise if it yields, you switch to the main thread only to immediately switch back.

Reply:
@sharwell Good call, fixed.